### PR TITLE
doc(website): Add how-to guides section to website

### PR DESCRIPTION
**Pull request by hermanschaaf**
This moves some of our how-to guide-style blog posts into their own section on the website, called "How-to Guides". I hope this will help with discoverability of the various interesting ways you can use CloudQuery data! It also helps to highlight where we still have some gaps in our documentation.

This is just a first attempt to make an iterative improvement. I imagine, as the section grows, we may need to add some different sections / tags / something else.

While looking over these I also decided to shorten some titles and do some light editing, so that it displays more nicely on the index page. Hopefully these minor edits are not too controversial.

### CODE REVIEW

**Comment by hermanschaaf** [comment]
Preview: https://cloudquery-qqw4m00r0-cloudquery.vercel.app/

File: website/components/HowToGuideIndex.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  frontMatter?: any;
+}
+
+function sortByTitle(a: Page, b: Page) {
+  return (
+     b.frontMatter?.title?.toLowerCase() < a.frontMatter?.title?.toLowerCase() ? 1 : -1
+  );
+}
+
+export function HowToGuideIndex({ more = "Read the guide" }) {
```
Comment (yevgenypats) [inline website/components/HowToGuideIndex.tsx:18]: Is this one necessary? i.e can't we use the same thing built-in theme as for the blog?

**Review by yevgenypats — APPROVED** [review]
Looks great. Only nit is if possible to avoid some of the custom frontend code and use the defaults somehow but we can do this later as it's not much code (for now)

File: website/components/HowToGuideIndex.tsx (hunk 1 above)
Comment (hermanschaaf) [reply website/components/HowToGuideIndex.tsx:18]: We probably can if we provide the blog index with some more options, but the ordering is different for how-to guides (they currently are ordered by title, not by date)

File: website/vercel.json [hunk 2]
```diff
@@ -144,6 +144,46 @@
     {
       "source": "/docs/plugins/sources/aws/multi_account",
       "destination": "/docs/plugins/sources/aws/multi-account"
+    },
+    {
+      "source": "/blog/cloud-asset-inventory-cloudquery-apache-superset",
+      "destination":"c/how-to-guides/loud-asset-inventory-cloudquery-apache-superset"
```
Comment (erezrokah) [inline website/vercel.json:150]:
```suggestion
      "destination":"/how-to-guides/cloud-asset-inventory-cloudquery-apache-superset"
```

Comment (erezrokah) [reply website/vercel.json:150]: giphy (1)

File: website/pages/how-to-guides/_meta.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "*": {
+    "//": "Disable sidebar and search for these pages",
+    "theme": {
+      "sidebar": false,
+      "typesetting": "article"
+    },
+    "searchable": false
+  },
+  "cloud-asset-inventory-cloudquery-apache-superset": {
```
Comment (erezrokah) [inline website/pages/how-to-guides/_meta.json:10]: What's the reason we need to explicitly set these?

File: website/components/HowToGuideHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+export function HowToGuideHeader() {
+  const config = useConfig();
+  const meta = config.frontMatter as PostMeta;
+
+  if (!meta) {
+    return null;
+  }
+
+  const image = `https://www.cloudquery.io/og-image/${meta.title}`;
```
Comment (erezrokah) [inline website/components/HowToGuideHeader.tsx:57]:
```suggestion
  const image = `/og-image/${meta.title}`;
```

We might want to use a relative URL so deploy previews point to it under their domain

File: website/components/HowToGuideIndex.tsx (hunk 1 above)
Comment (erezrokah) [reply website/components/HowToGuideIndex.tsx:18]: Can we remove the dates from the how to guides? I think that will ensure they get sorted by titles

File: website/pages/how-to-guides/_meta.json (hunk 3 above)
Comment (hermanschaaf) [reply website/pages/how-to-guides/_meta.json:10]: The reason is that I wanted the titles to be slightly different in the index than on the actual post. We have 5 posts that all start with `How to Build an Open Source Cloud Asset Inventory with` and the long titles made the index page look both repetitive and badly spaced. So in the index I called it `How Visualize CloudQuery data with...` to make it shorter but keep the original titles + url paths for SEO reasons.

File: website/components/HowToGuideHeader.tsx (hunk 4 above)
Comment (hermanschaaf) [reply website/components/HowToGuideHeader.tsx:57]: Cool, I'll update it here and in the Blog header where this was copied from

File: website/components/HowToGuideIndex.tsx (hunk 1 above)
Comment (hermanschaaf) [reply website/components/HowToGuideIndex.tsx:12]: Removed 👍

**Review by erezrokah — APPROVED** [review]

**Pull request merged**